### PR TITLE
refactor: Remove generic errors

### PR DESCRIPTION
**Pull request by KirilMihaylov**

### CODE REVIEW

File: contracts/lease/src/contract/endpoins.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ pub fn migrate(deps: DepsMut<'_>, _env: Env, _msg: MigrateMsg) -> ContractResult
     #[cfg(not(feature = "migration"))]
     let resp = versioning::update_software(deps.storage, version!(CONTRACT_STORAGE_VERSION))
         .and_then(response::response);
-    resp.or_else(|err| log_error(err, deps.api))
+    resp
```
Comment (gmanev7) [inline contracts/lease/src/contract/endpoins.rs:72]: what's the motivation ?

File: contracts/oracle/src/error.rs [hunk 2]
```diff
@@ -8,9 +8,52 @@ use sdk::cosmwasm_std::{Addr, StdError};
 
 #[derive(Error, Debug, PartialEq)]
 pub enum ContractError {
+    #[cfg(feature = "testing")]
     #[error("[Oracle] [Std] {0}")]
     Std(#[from] StdError),
 
+    #[error("[Oracle; Stub] Failed to query configuration! Context: {0}")]
```
Comment (gmanev7) [inline contracts/oracle/src/error.rs:15]: [everywhere] Please replace `Context` with `Cause`

File: contracts/oracle/src/error.rs [hunk 3]
```diff
@@ -8,9 +8,52 @@ use sdk::cosmwasm_std::{Addr, StdError};
 
 #[derive(Error, Debug, PartialEq)]
 pub enum ContractError {
+    #[cfg(feature = "testing")]
```
Comment (gmanev7) [inline contracts/oracle/src/error.rs:11]: let's minimize conditional compilation and avoid test-custom errors

File: contracts/lease/src/contract/endpoins.rs (hunk 1 above)
Comment (KirilMihaylov) [reply contracts/lease/src/contract/endpoins.rs:72]: That must have happened during the rebase as it was really old at this point.

File: contracts/oracle/src/error.rs (hunk 3 above)
Comment (gmanev7) [reply contracts/oracle/src/error.rs:11]: remove the Std variant as you did elsewhere and `.unwrap()`

File: contracts/oracle/src/error.rs (hunk 2 above)
Comment (gmanev7) [reply contracts/oracle/src/error.rs]: by "everywhere" I meant any occurrence in this PR including marketprice errors

File: packages/versioning/src/lib.rs [hunk 4]
```diff
@@ -94,18 +94,11 @@ pub fn initialize(storage: &mut dyn Storage, version: Version) -> StdResult<()>
...
 }
 
-pub fn update_software<ContractError>(
-    storage: &mut dyn Storage,
-    new: Version,
-) -> Result<ReleaseLabel, ContractError>
-where
-    StdError: Into<ContractError>,
-{
+pub fn update_software(storage: &mut dyn Storage, new: Version) -> Result<ReleaseLabel, StdError> {
```
Comment (gmanev7) [inline packages/versioning/src/lib.rs:97]: just noticed that the return error is not consistent with `update_software_and_storage`. Update it as well

File: packages/versioning/src/lib.rs [hunk 5]
```diff
@@ -94,33 +94,45 @@ pub fn initialize(storage: &mut dyn Storage, version: Version) -> StdResult<()>
...
+where
+    MapErrorFunctor: Fn(StdError) -> ContractError,
+{
     load_version(storage)
-        .and_then(|current| release::allow_software_update(&current, &new))
-        .and_then(|()| save_version(storage, &new))
+        .map_err(&map_error)
+        .and_then(|current| release::allow_software_update(&current, &new).map_err(&map_error))
+        .and_then(|()| save_version(storage, &new).map_err(map_error))
         .map(|()| release::label())
```
Comment (gmanev7) [inline packages/versioning/src/lib.rs:109]: is not possible to `.map_err` only once at the end?

Comment (KirilMihaylov) [reply packages/versioning/src/lib.rs:109]: That's a very good point. Thank you!

**Review by gmanev7 — APPROVED** [review]

**Pull request merged**